src/fsharp/FSharp.Core/control.fs

```diff
@@ -1462,6 +1462,50 @@ namespace Microsoft.FSharp.Control
                             |> unfake);
                     FakeUnit))
 
+        static member Choice(computations : Async<'T option> seq) : Async<'T option> =
+            unprotectedPrimitive(fun args ->
+                let result =
+                    try Choice1Of2 <| Seq.toArray computations
+                    with exn -> Choice2Of2 <| ExceptionDispatchInfo.RestoreOrCapture(exn)
+
+                match result with
+                | Choice2Of2 edi -> args.aux.econt edi
+                | Choice1Of2 [||] -> args.cont None
+                | Choice1Of2 [|P body|] -> body args
```

Comment:
I think just remove this case.  It's better to run all operations consistently through the thread pool.